tests/test_storage.py

```diff
@@ -25,87 +25,128 @@ def _make_local_files(files):
 
-    zg = access(store, mode='w')
-    zg['foo'] = data
-    assert zarr.open(DEFAULT_ZARR_STORE(store), mode='a') == zg
+    zg = access(store, mode="w")
+    zg["foo"] = data
+    assert zarr.open(DEFAULT_ZARR_STORE(store), mode="a") == zg
 
 
-def test_accessing_group_zarr_n5():
-    store = tempfile.mkdtemp(suffix='.n5')
+def test_access_group_zarr_n5():
+    store = tempfile.mkdtemp(suffix=".n5")
     atexit.register(shutil.rmtree, store)
-    data = np.zeros(100, dtype='uint8') + 42
-    zg = zarr.open(store, mode='a')
-    zg.attrs.update({'foo': 'bar'})
-    zg['foo'] = data
+    data = np.zeros(100, dtype="uint8") + 42
+    zg = zarr.open(store, mode="a")
+    zg.attrs.update({"foo": "bar"})
+    zg["foo"] = data
 
-    assert dict(access(store, mode='r').attrs) == {'foo': 'bar'}
-    assert np.array_equal(access(store, mode='r')['foo'][:], data)
+    assert dict(access(store, mode="r").attrs) == {"foo": "bar"}
+    assert np.array_equal(access(store, mode="r")["foo"][:], data)
 
 
-def test_accessing_precomputed():
-    store = tempfile.mkdtemp(suffix='.precomputed')
+def test_access_precomputed():
+    store = tempfile.mkdtemp(suffix=".precomputed")
     atexit.register(shutil.rmtree, store)
-    key = 's0'
-    data = np.random.randint(0, 255, size=(10, 10, 10), dtype='uint8')
+    key = "s0"
+    data = np.random.randint(0, 255, size=(10, 10, 10), dtype="uint8")
     resolution = [1.0, 2.0, 3.0]
     chunks = (2,) * data.ndim
 
-    arr_out = access_precomputed(store,
-                                 key,
-                                 mode='w',
-                                 array_type='image',
-                                 dtype='uint8',
-                                 shape=data.shape,
-                                 num_channels=1,
-                                 resolution=resolution,
-                                 encoding='raw',
-                                 chunks=chunks)
-    arr_out[:] = data.reshape(*data.shape,1)
-
-    arr_in = access_precomputed(store, key, mode='r')
+    arr_out = access_precomputed(
+        store,
+        key,
+        mode="w",
+        array_type="image",
+        dtype="uint8",
+        shape=data.shape,
+        num_channels=1,
+        resolution=resolution,
+        encoding="raw",
+        chunks=chunks,
+    )
+    arr_out[:] = data.reshape(*data.shape, 1)
+
+    arr_in = access_precomputed(store, key, mode="r")
     assert np.all(np.array(arr_in[:]) == data.reshape(*data.shape, 1))
 
     darr = precomputed_to_dask(os.path.join(store, key), chunks=(2, 2, 2))
     assert (darr.compute() == data).all
 
 
+def test_access_array_h5():
+    key = "s0"
+    data = np.random.randint(0, 255, size=(10, 10, 10), dtype="uint8")
+    attrs = {"resolution": "1000"}
+    chunks = (2,) * data.ndim
+
+    with tempfile.TemporaryFile(suffix=".h5") as store:
+        arr = access_h5(store, key, chunks=chunks, data=data, attrs=attrs, mode="w")
```

Comment:
Might want to test `compression="gzip", compression_opts=1` here as well since that is a built-in filter.

Reply:
Also include a test for the bulitin `shuffle` filter.

Reply:
i decided not to add tests for these parts of the h5py.dataset API, since `access_h5` is such a light wrapper

Reply:
If this "light wrapper" did not require a comprehensive list of the upstream library's keyword arguments, I would agree.